### DISCUSSION TITLE
new features: dimmer and web auth

### DISCUSSION THREAD

**Discussion by guiumb** [Claus-E/teams2tasmota#3] [Ideas]
Hi Claus,

Great job ! IMHO it is very nice project !

Could I suggest a couple new features to add:

- insert in xlm config a dimmer value for each state in order to customize led brightness (https://tasmota.github.io/docs/Commands/#light)
- insert in xlm user credential for web authentication on Tasmota web gui (https://tasmota.github.io/docs/Commands/#with-web-requests)

I hope that these new fetaures should be fine for many other guys.

many thanks
Umberto

**Comment by Claus-E** [comment]
Hi Umberto,

thank you for the complimentes!

I know the Dimmer command, but i can't see the advantage? You can reach the same by modifing the RGB Values. Or did you want to have one central Dimmervalue for all Colors?

You can use the credential for web authentication with out a change of the Software. Simply edit the xml. But this is overwritten when you change the color via the dialog. And you have to edit all color-strings.
A central place for this is possible.

Greetings Claus

**Reply by guiumb** [reply]
Hi Claus,

I think that a central dimmer value for all colors should be a good idea: whit dimmer functionality I can select the right amount of light emitted by device in order to have a clear visibility status without "flashing" collegues on near office desk !

About web auth should be nice to have a config central parameter: it will permit changing colors without any functionaluty intraction with web gui.

Thanks for your time

Umberto

**Reply by Claus-E** [reply]
Hi Umberto,

the new Version with your features is online!

I hope you are happy with it. Feel free to send comments if not.

Greetings Claus

**Comment by guiumb** [comment]
Hi Claus,

I'm very happy and thankful you were able to add those new features in such a short time! I hope your work will be useful to many more people other than me; again, I really appreciate the effort you put in it, and how you managed to implement those suggestions I gave you. 

Umberto
